kong/plugins/ai-prompt-guard/handler.lua

```diff
@@ -26,16 +26,21 @@ local execute do
   -- @tparam table request The deserialized JSON body of the request
   -- @tparam table conf The plugin configuration
   -- @treturn[1] table The decorated request (same table, content updated)
-  -- @treturn[2] nil
```

Comment:
This line shouldn't be removed, it breaks the docs format

Reply:
there's another `treturn[2]` below, this is a typo i think

Reply:
each line is a (typed) value, the numbers indicate the return "options";

1. a valid result, 1 value, being a table
2. an error which returns 2 values:
    - a nil value
    - an error message

so 3 lines in total

see first example here: https://lunarmodules.github.io/ldoc/examples/multiple.lua.html

Reply:
I see, in other places however we use following patterns:
```lua
-- @treturn table          something
-- @treturn string|nil    error
```
should we align with that instead?

Reply:
updated with `@treturn string|nil`

Reply:
There's PDK and other plugins uses this pattern, a took a snippet below:
```
kong/db/schema/topological_sort.lua:-- @treturn nil|string nil if the schemas were sorted, or a message if a cycle was found
kong/db/declarative/init.lua:-- @treturn nil|string error message, only if error happened
kong/db/declarative/init.lua:-- @treturn nil|table err_t, only if error happened
kong/db/declarative/init.lua:-- @treturn nil|string error message, only if error happened
kong/db/declarative/init.lua:-- @treturn nil|table err_t, only if error happened
kong/db/declarative/init.lua:-- @treturn nil|string error message if error
kong/db/declarative/init.lua:-- @treturn nil|table err_t if error
kong/tools/stream_api.lua:-- @treturn nil|string error
kong/tools/stream_api.lua:-- @treturn nil|string error   an error string
kong/tools/stream_api.lua:-- @treturn nil|string error  an error string, returned for protocol or socket I/O failures
kong/tools/stream_api.lua:-- @treturn nil|string   error
kong/tools/mime_type.lua:-- @treturn nil|nil|nil Invalid mime-type
kong/workspaces/counters.lua:-- @treturn nil|string error
kong/openid-connect/claims.lua:-- @return status(nil|boolean), error(string)
kong/openid-connect/claims.lua:-- @return status(nil|boolean), error(string)
kong/openid-connect/claims.lua:-- @return status(nil|boolean), error(string)
kong/openid-connect/claims.lua:-- @return status(nil|boolean), error(string)
kong/pdk/client/tls.lua:  -- @treturn nil|err Returns `nil` if successful, or an error message if it fails.
kong/pdk/client/tls.lua:  -- @treturn nil|err Returns `nil` if successful, or an error message if it fails.
kong/pdk/client/tls.lua:  -- @treturn nil|err Returns `nil` if successful, or an error message if it fails.
kong/pdk/request.lua:  -- @treturn nil|string An error message.
kong/pdk/client.lua:  -- @treturn nil|err `nil` if successful, or an error message if it fails.
kong/pdk/client.lua:  -- @treturn nil|err `nil` if successful, or an error message if it fails.
```

Right now, only the newly added ai-prompt-guard and ai-azure-content-safety plugin uses the new pattern. I'm not sure how we have two different patterns and our docs are rendering fine with the PDK ldoc.
Anyway, I'm going to just revert this change. I do not use ldoc locally so I don't have clue how to verify it.

Reply:
I don't know who wrote that, but most of that is wrong.

Reply:
would you take another look at this PR? @Tieske